app/models/manageiq/providers/amazon/cloud_manager.rb

```diff
@@ -122,55 +122,48 @@ def self.default_blacklisted_event_names
 
   def self.params_for_create
     @params_for_create ||= {
-      :title  => "Configure AWS",
       :fields => [
         {
-          :component  => "text-field",
-          :name       => "name",
-          :label      => "Name",
-          :isRequired => true,
-          :validate   => [{:type => "required-validator"}]
-        },
-        {
-          :component  => "text-field",
-          :name       => "zone_name",
-          :label      => "Zone",
-          :isRequired => true,
-          :validate   => [{:type => "required-validator"}]
-        },
-        {
-          :component  => "text-field",
+          :component  => "select-field",
           :name       => "provider_region",
-          :label      => "Region",
+          :label      => _("Region"),
           :isRequired => true,
-          :validate   => [{:type => "required-validator"}]
+          :validate   => [{:type => "required-validator"}],
+          # TODO: we will need some kind of dynamic loading component that's able to load stuff based on static information
         },
         {
-          :component  => "text-field",
-          :name       => "endpoints.default.userid",
-          :label      => "Access Key",
-          :isRequired => true,
-          :validate   => [{:type => "required-validator"}]
+          :component => 'sub-form',
+          :name      => 'endpoints',
+          :title     => _("Endpoints"),
+          :fields    => [
+            {
+              :component => "text-field",
+              :name      => "<TBD>", # TODO: maybe proxy uri? (default_url in the old world)
```

Comment:
This is a big big :question: for me @agrare :confused:

Reply:
This is the stuff I was able to find, it is being sent as an argument to `validate_credentials_task`:
```ruby
          uri = URI.parse(WEBrick::HTTPUtils.escape(params[:default_url]))
          [
            user,
            password,
            :EC2,
            params[:provider_region],
            ems.http_proxy_uri,
            true,
            uri,
            {:assume_role => params[:default_assume_role].presence},
          ]
```

So I guess it becomes the 7th argument of [`raw_connect`](https://github.com/ManageIQ/manageiq-providers-amazon/blob/01feee535cc84e6da97590ae1b31145416ea8815/app/models/manageiq/providers/amazon/manager_mixin.rb#L79) which is in the provider.

Reply:
Okay so that's if you have a different endpoint from the normal one e.g. gov cloud?  I'm wondering how that's different from using a proxy

Reply:
I don't even know what gov cloud is :sweat_smile: so good question.

Reply:
Well if we support it now we shouldn't change that, its just weird :)

Reply:
Ok, how should I name the field? So your validation/add endpoints can handle it.

Reply:
It should just be called `"url"` since that's what it is called on the endpoint table

Reply:
Okay more confused, we don't pass that in when we use the instance method [`#connect`](https://github.com/ManageIQ/manageiq-providers-amazon/blob/master/app/models/manageiq/providers/amazon/manager_mixin.rb#L20-L33) so even if it validated with it, it would never be used (?!?) (:scream:)

Reply:
`endpoints.default.url`